src/kv/tx.h

```diff
@@ -54,7 +55,9 @@ namespace kv
       if (read_version == NoVersion)
       {
         // Grab opacity version that all Maps should be queried at.
-        read_version = m.get_store()->current_version();
+        auto [t, rv] = m.get_store()->current_term_and_version();
+        term = t;
```

Comment:
Just trying to make sure I understand this properly: we're setting up a long-range invariant here, that the term when we create our first `TxView` will match the term when our transaction is committed. We don't explicitly check/enforce this. In most situations I think this holds (a term change will cause a transaction to fail on commit, or to be discarded from `pending_txs`).

The potential hole I see is if we've put this serialised transaction into a batch, and we're just about to call `replicate` with it, but there's an election and the term changes. If replicate can succeed, we'll report to the user that this transaction was `2.5`, but we'll replicate it to other nodes in term 3. I think this is still fine - usually the node is no longer the `Leader` so replicate fails, and even if it remains `Leader` it has probably rolled back so will not accept this batch as the next index. But is this narrow case possible, where we have a locally committed write about to be replicated, there's an election but we become `Leader`, no other writes happen, and so we replicate this transaction in the wrong term?

Reply:
> We don't explicitly check/enforce this.

I assume that this will be enforced once the transaction term is passed to `store->commit()` (see `TODO` on line `209`).

Reply:
@eddyashton @jumaffre I agree this is a hole, and the validation I am thinking of is that we should pass the term to store->commit(), and that under the version lock, we check the term of inbound commits versus current term. If they don't match, there's been a term change and we drop the tx (and return NO_REPLICATE).

Thoughts?

Reply:
> ... versus current term.

Is that retrieved from the consensus `r` directly? If so, this was my understanding too.

Reply:
That explicit validation sounds good, but I'm not sure it catches the final hole I described. We hold the version lock while building the batch, but deliberately release it before calling `replicate()`. So we can get a term match while building the batch, but that doesn't mean the term is still 'correct' when we call `replicate()`.

Reply:
Yes, although as discussed we don't need to store the term per-batch because we can:

1. reject Store::commit()s for a term that does not match current
2. send the current term to replicate(), which can in turn do the same, and yied NO_REPLICATE